src/main/java/org/jvnet/hudson/test/RealJenkinsRule.java

```diff
@@ -744,8 +870,14 @@ public void startJenkins() throws Throwable {
                     + ",suspend=" + (debugSuspend ? "y" : "n")
                     + (debugPort > 0 ? ",address=" + httpListenAddress + ":" + debugPort : ""));
         }
-        argv.addAll(javaOptions);
+        if(!bootClasspathFiles.isEmpty()) {
+            String pathSeparator = Functions.isWindows() ? ";" : ":";
+            String fileList = String.join(pathSeparator, bootClasspathFiles.stream()
+                    .map(file -> FilenameUtils.separatorsToSystem(file.getAbsolutePath())).toList());
```

Comment:
this makes my head hurt.  bootClasspathFiles is a list of files (could just be paths, see above)
and you are using a stream to create a list to then join.. but the stream can do it all.
Additionally I am not sure why FilenameUtils.separatorsToSystem is being used here - as the File/Path should already be correct as the path comes from the system and has been made absolute)?

For example

```java
String fileList = bootClasspathFiles.stream().map(File::getAbsolutePath).collect(Collectors.joining(File.pathSeparator));
```

Reply:
this hurt my head a lot to get this working on windows :)
maybe this would work on windows I can have a try